cel/cos_tlv.go

```diff
@@ -37,3 +57,17 @@ func (c CosTlv) GenerateDigest(hashAlgo crypto.Hash) ([]byte, error) {
 	}
 	return hash.Sum(nil), nil
 }
+
+// NewCosTlvFromTlv constructs a CosTlv from a TLV. It will check for the correct COS event
+// type, and unmarshal the nested event.
+func NewCosTlvFromTlv(tlv TLV) (CosTlv, error) {
```

Comment:
why do we need this? And if we do, should we export it?

Reply:
I think yes, we need a way to parse a regular TLV (record.content) into a CosTlv, after then we can read its content

Reply:
Oh I see, thanks! I got confused reading this, since I was thinking higher level: something that goes through a CEL and per supported event type tried to convert to a content specific value (`CEL`, `PCCLIENT_STD`, `COS_TYPE`, etc.). Regardless, we need this.

Let's turn this into a receiver function on TLV.

Reply:
Yep, changing to ParseToCosTlv()